src/main/kotlin/net/djvk/fireflyPlaidConnector2/transactions/TransactionConverter.kt

```diff
@@ -186,17 +200,29 @@ class TransactionConverter(
          * Don't pass in [plaidUpdatedTxs] here because we're not going to try to update transfers for now
          *  because it's more complexity than I want to deal with, and I haven't seen any Plaid updates in the wild yet
          */
-        val (singles, pairs) = sortByPairs(plaidCreatedTxs + transferCandidateExistingFireflyTxs, accountMap)
-        logger.debug("${::convertPollSync.name} call to ${::sortByPairs.name} received ${singles.size} singles and ${pairs.size} pairs")
+        val (singles, pairs) = transferMatcher.match(
+            PlaidFireflyTransaction.match(plaidCreatedTxs, transferCandidateExistingFireflyTxs, accountMap)
```

Comment:
This adds net-new behavior in terms of `PlaidFireflyTransaction.match`'s id matching behavior. What's the intent there? Is that solving a new problem?

Reply:
Ah, apologies for not explaining this better in the description. In short, this new ID matching behavior is added for idempotency.

I'm new to this project, and to Plaid, but based on my understanding PolledSyncRunner fetches FF transactions and supplies them here along with the most recent Plaid transactions. Normally PolledSyncRunner uses a fresh cursor every execution, so normally we should only see each transaction once (ignoring the mythical "update" for now). In that world where everything works fine, the old code is great.

However, if we were to fail to write the new cursor after performing a sync (maybe the disk is full, awkwardly-timed power failure, etc), we could fetch again using the old cursor and I think that would mean that we would see a new "created" transaction again. By preemptively looking for matches between the Plaid and Firefly transactions we can achieve more predictable results when this happens.

I think this may also have other benefits, such as making it easier to load existing FF transactions in batch sync mode for the purpose of matching transfers (for example, when adding a new financial institution to Plaid that already has transfers that have been loaded into FF from an already-sync'd institution).